extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmConfigPersistenceUnit.java

```diff
@@ -310,13 +310,52 @@ interface HibernateOrmConfigPersistenceUnitDialect {
          *
          * E.g. `MyISAM` or `InnoDB` for MySQL.
          *
+         * @deprecated Use {@code mysql.}{@linkplain MySQLDialectConfig#storageEngine storage-engine} instead
+         *
          * @asciidoclet
          */
         @WithConverter(TrimmedStringConverter.class)
+        @Deprecated
         Optional<String> storageEngine();
 
+        /**
+         * Configuration specific to the Hibernate ORM {@linkplain org.hibernate.dialect.CockroachDialect}
+         */
+        CockroachDialectConfig cockroach();
+
+        /**
+         * Configuration specific to the Hibernate ORM {@linkplain org.hibernate.dialect.HANADialect}
+         */
+        HANADialectConfig hana();
+
+        /**
+         * Configuration specific to the Hibernate ORM {@linkplain org.hibernate.dialect.MySQLDialect}
+         */
+        MySQLDialectConfig mysql();
```

Comment:
Note we have use dedicated configuration for MariaDB in Quarkus, e.g. there is a `db-kind` named `mariadb`, distinct from `mysql`. It would probably make sense to do the same here?

FWIW you *might* be able to use inheritance with config groups if you want to avoid duplicating code... I think Guillaume added that support in his recent changes to the config annotation processor. We'll need to double checked by having a look at the [resulting documentation](https://quarkus-pr-main-43396-preview.surge.sh/version/main/guides/hibernate-orm#hibernate-configuration-properties), and of course through tests.

Reply:
I could just add `MySQLDialectConfig mariadb();`, no?

Reply:
True, for now this would work.